CONTRIBUTING.md

```diff
@@ -26,110 +47,274 @@ Before we get started, here are a few things we expect from you (and that you sh
+
+Note:
+> Currently since we have been refactoring the app, the upcoming hardening sprints will be focused on adding tests to existing features. Our plan is to go app by app and build test
+
+During this period we make a build available on the insiders channel. We will monitor incoming issues from this release, fix any critical bugs that arise, and then produce a final stable release for the monthly iteration.
+
+## QA testing
+We perform QA testing when features are merged from `develop` to `staging`. We use automated end-to-end testing to provide a strong foundation for functional testing.
+
+---
+
+# How we manage our code
+
 **Source control:** Git
- 
+
+**Source code:** All our repositories are stored on [GitHub](https://github.com/hikaya-io).
+
 **Branches:**
- 
-We have 3 main branches in our repo:
-- `Dev` - All bug fixes and enhancements
-- Feature branches (i.e. `ACT-001`) - All feature branches must be named like: ACT-001 (where ACT-001 is the issue number on GitHub)
-- `Staging` - This is our QA testing environment
-- `Master` - Clean code in production environment
- 
-**Hot-fixes & bug-fixes**
- 
-- All pull requests should be based off of the `Dev` branch.
-- All branch names should follow the issue number such as: `ACT-001`. This should be the Github issue number.
- 
-**Commits (Smart Commits, only for Jira)**
- 
-- For us to be able to track and link our issues on Github, we use smart commits (This is for issues tracked in Jira). Below is an example of smart commit:
-  - ACT-001:- Update Activity UI based on styling guide
-  - ACT-001:- is the issue number on Jira
-    - “Update Activity UI based on styling guide” Is the commit body. This should be as descriptive as possible.
- 
-**Pull Requests**
- 
+
+We use 3 main branches in all our repositories :
+
+`Dev` (or `Develop`) - All bug fixes, defects, and tasks.<br>
+`Feature branches` (i.e. ACT-001) - All feature branches must be named like: ACT-001 (where ACT-001 is the issue number on GitHub).<br>
+`Staging` - This is our QA testing environment<br>
+`Master` - Clean code in production environment<br>
+
+> **Note:**
+> * All pull requests should be based off of the `develop` branch.
+> * All branch names should follow the issue number such as: ACT-001. The numbering should follow the auto-generated GitHub issue number.
+
+## Commits
+
+We use Github to track **Issues** and **Pull Requests* for all our work. When making commits, please ensure that you have create a separate branch off of the main `develop` branch before you begin making changes to the source code. Feel free to make commits as often as possible to your branch. If the issue you are working on has been open for a few days, consider pulling in the latest changes to keep in up to date with the main `develop` branch before you create a Pull Request. This will help you avoid merge conflicts when raising your Pull Request.
+
```

Comment:
Can we add a subsection here on template to write commits?
Suggestion:

### Commit message format
To ensure commit messages are readable when going through project history, we use a common commit message format. The message format by [Angular](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines) is very comprehensive and we find it useful to follow.

Each commit message consists of a header, a body and a footer. The header is mandatory with the body and the footer as optional given the size of the commit. The header has a special format that includes a type, a scope, and a subject, where the scope of the header is optional.

```
<type>(<scope>): <subject>
<BLANK LINE>
<body>
<BLANK LINE>
<footer>
```

Please refer to the [Angular guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines) for more details on the format.